### PR TITLE
Updating REST Transport Documentation

### PR DESCRIPTION
**Pull request by oslynn**
This PR updated the REST Transport Documentation to reflect the new modified REST Transport

Test Steps:
1. *Include steps to test these changes*

## Changes
- *Include a comprehensive list of changes in this PR*
- *(For web UI changes) Include screenshots/video of changes*

### Testing
- [ ] Tested locally?
- [ ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [ ] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Fixes #issue
- https://app.zenhub.com/workspaces/engagement-6166edbd409257001e09f1b2/issues/gh/cdcgov/prime-reportstream/12680

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### CODE REVIEW

File: prime-router/docs/onboarding-users/transport/rest.md [hunk 1]
```diff
@@ -83,7 +83,33 @@ The receiver's RESTTransport includes the following fields:
     - authType:       Authentication type i.e two-legged.  It is default to null (API shared key)
     - tlsKeystore:    The name for the credential manager to get the JKS used in TLS/SSL
     - parameters:     The map of parameters to be sent in the message (use with two-legged OAuth.  See ETOR RESTTransport)
+    - authHeaders:    The key:value of authentication header
+        . Authentication-Type:  Can be as following
+            "Basic Auth" - Authenticates with string "Basic + base64(username+password)" to authentication header
+            "username/password" - Authenticates with json body { "username": "<username>", "password": "<password>" }
+            "email/password" - Authenticates with json body { "email": "<username>", "password": "<password>" }
```
Comment (adegolier) [inline prime-router/docs/onboarding-users/transport/rest.md:90]:
```suggestion
            "email/password" - Authenticates with json body { "email": "<email>", "password": "<password>" }
```

File: prime-router/docs/onboarding-users/transport/rest.md [hunk 2]
```diff
@@ -83,7 +83,33 @@ The receiver's RESTTransport includes the following fields:
...
+        . Content-Type: Can be as following
+            "application/json" - Authentication body as json
+            "text/plain" - Authentication body as text/plain
+            "Subscription" - Given to us by STLT (It is as plublic key and not secrete)
```
Comment (adegolier) [inline prime-router/docs/onboarding-users/transport/rest.md:94]:
```suggestion
            "Subscription" - Given to us by STLT (It is as public key and not secret)
```

File: prime-router/docs/onboarding-users/transport/rest.md [hunk 3]
```diff
@@ -83,7 +83,33 @@ The receiver's RESTTransport includes the following fields:
...
+        . ExpectSuccess: Set to true for expecting success and if not it will throw exception
+        . Host: Set to host domain name.
     - headers:        The map of headers to be sent in the message
+        . Host: Set to host domain name.
+        . Content-Type: Can be as following
+            "application/json" - Content/message body as json
+            "text/plain" - Content/message body as text/plain
+            "multipart/form-data"  - Content/message body as multipart/form-data or file
+            "Subscription" - Given to us by STLT (It is as plublic key and not secrete)
```
Comment (adegolier) [inline prime-router/docs/onboarding-users/transport/rest.md:103]:
```suggestion
            "Subscription" - Given to us by STLT (It is as public key and not secret)
```

File: prime-router/docs/onboarding-users/transport/rest.md [hunk 4]
```diff
@@ -83,7 +83,33 @@ The receiver's RESTTransport includes the following fields:
...
+        . Key: If Content-Type is "multipart/form-data", it Can be as following
+            "payload" - Set "multipart/form-data" key value to payload (uses by NY-PHD)
```
Comment (adegolier) [inline prime-router/docs/onboarding-users/transport/rest.md:105]:
```suggestion
            "payload" - Set "multipart/form-data" key value to payload (used by NY-PHD)
```

File: prime-router/docs/onboarding-users/transport/rest.md [hunk 5]
```diff
@@ -83,7 +83,33 @@ The receiver's RESTTransport includes the following fields:
...
+            "files" - Set "multipart/form-data" key value to file (uses by LA-PHL/Natus).
```
Comment (adegolier) [inline prime-router/docs/onboarding-users/transport/rest.md:106]:
```suggestion
            "files" - Set "multipart/form-data" key value to file (used by LA-PHL/Natus).
```

**Review by adegolier — APPROVED** [review]
A few minor typo corrections, but otherwise lgtm

File: prime-router/docs/onboarding-users/transport/rest.md (hunk 1 above)
Comment (oslynn) [reply prime-router/docs/onboarding-users/transport/rest.md:90]: Thank you @adegolier

File: prime-router/docs/onboarding-users/transport/rest.md [hunk 6]
```diff
@@ -83,7 +83,34 @@ The receiver's RESTTransport includes the following fields:
...
+        . Host: Set to host domain name.
+        . Content-Type: Can be as following
+            "application/json" - Content/message body as json
+            "text/plain" - Content/message body as text/plain
+            "multipart/form-data"  - Content/message body as multipart/form-data or file
+            "Subscription" - Given to us by STLT (It is as public key and not secrete)
+        . Key: If Content-Type is "multipart/form-data", it Can be as following
+            "payload" - Set "multipart/form-data" key value to payload (used by NY-PHD)
+            "files" - Set "multipart/form-data" key value to file (used by LA-PHL/Natus).
+        . File-Name: Set customer message's file name.  If the name includes substring "withdate", it will include datetime with the name.
```
Comment (victor-chaparro) [inline prime-router/docs/onboarding-users/transport/rest.md:107]: Is this still needed?

**Review by victor-chaparro — CHANGES_REQUESTED** [review]

**Comment by oslynn** [comment]
Used the existing Report.formFileName to generate the filename.  Therefore, we don't need to update the document.

**Pull request closed without merging**